src/main/java/seedu/modtrek/model/DegreeProgressionData.java

```diff
@@ -0,0 +1,74 @@
+package seedu.modtrek.model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import seedu.modtrek.model.module.Module;
+import seedu.modtrek.model.module.UniqueModuleList;
+
+/**
+ * DegreeProgressionData holds all the data for an overview of Degree Progression.
+ */
+public class DegreeProgressionData {
+
+    // Currently for cohort 2122
+    public static final int TOTALCREDIT = 160;
+    public static final Map<String, Integer> REQUIREMENTCREDITS = Map.of(
+            "UNIVERSITY_LEVEL_REQUIREMENTS", 16,
+            "COMPUTER_SCIENCE_FOUNDATION", 36,
+            "COMPUTER_SCIENCE_BREADTH_AND_DEPTH", 40,
+            "IT_PROFESSIONALISM", 12,
+            "MATHEMATICS_AND_SCIENCES", 16,
+            "UNRESTRICTED_ELECTIVES", 40);
+    private int currentCredit = 0;
+    private HashMap<String, Integer> completedRequirementCredits = new HashMap<>();
+    private float cumulativePoints = 0;
+    private float gpa;
+
+    private DegreeProgressionData() {}
+
+    /**
+     * Factory method to generate the data needed to show progress. Progress
+     * is calculated from the user's current modlist.
+     *
+     * @param modList
+     * @return
+     */
+    public static DegreeProgressionData generate(UniqueModuleList modList) {
```

Comment:
Just to confirm, the UniqueModuleList contains only modules with grade and tags right? As mentioned, without either of those two, I suggest that we can assume those modules are incomplete / planned by user, and hence should not be taken into account in degree progression / CAP calculation.

Reply:
Right, I missed out the incomplete modules here. `modList` here should have all the modules.

Reply:
Okay